### ISSUE TITLE
{{annotations.length}} works in main template but not output path file name template

### ISSUE THREAD

**Issue by KateNave** [mgmeyers/obsidian-zotero-integration#393]
Hello, 

I would like to only create literature notes for pdfs that do have some annotations, so ignoring those that are unannotated. To do this I'm attempting to use {% if annotations.length > 0%}... This works fine within the main template for the literature note, but for some reason in the template settings for the output path file name, including annotations.length creates an error. If I just get it to print {{annotations.length}} here, then it always prints 0, no matter how many annotations the file has. 

I cannot figure out why this would work in the main template, but not the output path one and any advice on this would be really appreciated - thanks!

**Comment by mgmeyers** [comment]
Yeah, unfortunately annotations are not accessible to the output path templates. This is because the output path has to be computed before extracting annotations from PDFs. I'll have to think on this a bit to see if there's a good way to only create notes for items with annotations.

**Comment by KateNave** [comment]
Ah, that makes sense - thanks :)

**Comment by KateNave** [comment]
In case anyone else is looking to set things up for only importing annotated inputs, I thought I'd share the solution I found. 

1) This add-on for Zotero allows you to see if your items have annotations or not https://github.com/Dominic-DallOsto/zotero-annotations-count/tree/v0.2.4. 
2) Going to "classic view" in the Zotero search bar opens up the main Zotero view. If you import multiple sources, you can sort these by the "number of annotations" column to select only those that have annotations
3) To import only updated annotations each time, the Zotero add-on "Actions and Tags" allows you to auto-tag any items when you add a new annotation (under the "append annotation" option). I then created a collection using smart search including only items with this tag, import everything in this collection each time, then go to Zotero and manually remove the "NewlyAnnotated" Tag

Hope that helps!
